### ISSUE TITLE
option to skip dropdown after adding an item

### ISSUE THREAD

**Issue by sixmo** [Gbuomprisco/ng2-tag-input#222]
We'd like an option that prevents opening the dropdown again after adding or removing an item. Specifically for Tags accepting only items from an autocomplete.

Thanks heaps!!

**Comment by 4nandP** [comment]
I was just about to ask for this myself. 
**+1 for this feature.**

**Comment by Gbuomprisco** [comment]
Submitting a change of behaviour. Just FYI the option causing that is `showDropdownIfEmpty` (I'll make 2 separate examples as I understand the confusion here)

**Comment by Gbuomprisco** [comment]
Hi @sixmo and @4nandP,

this should now be OK on the new version. It will still show up the dropdown after you remove a tag, it was a bit tricky, but I'll try adjusting that too. FYI there is no new option, it is the default behaviour.

**Comment by szanella** [comment]
I'm experiencing a problem that might be somewhat related to this one.
I'm using the tag-input with the showDropdownIfEmpty option and the following issue happens: I focus the input, the dropdown shows up (along with its backdrop); now if I select an element, the dropdown and backdrop will disappear, for the backdrop to then suddenly appear (without the dropdown) and cover the whole screen, forcing me to click again to use the interface.

This is my setup:
<tag-input
      [placeholder]="''"
      [secondaryPlaceholder]="''"
      [name]="filter.key"
      [ngModel]="filter.value"
      (ngModelChange)="updateFilter($event)"
      [onlyFromAutocomplete]="true"
      [ripple]="false"
      (onAdd)="onAdd($event)"
      (onBlur)="onBlur($event)">
      <tag-input-dropdown [showDropdownIfEmpty]="true"
                          [autocompleteItems]="filter.options">
      </tag-input-dropdown>
    </tag-input>

I saw that part of the changes of the commit that solved the original issue of this thread were removed in subsequent commits.
What might be causing this?

**Comment by szanella** [comment]
On a side note on my previous comment, the issue does not happen upon removing an element, only after adding one.

Also, to be more specific about the code removal, I noticed that, at line 624 of modules/components/tag-input/tag-input.ts, the parameter named _displayAutocomplete_ is completely unused, while it was in the original commit to solve this issue.

**Comment by Gbuomprisco** [comment]
Hi @szanella,

Strange behaviour, but looking at your code I found it odd you use one-time data binding but use ngModelChange. Have you tried using 2-way binding instead? I am not sure if the subsequent cycle following updateFilter might cause that.

**Comment by szanella** [comment]
@Gbuomprisco thanks for the prompt response!
The reason I'm using the ngModelChange output is that my updateFilter method does several things besides simply updating the bound value.
Unfortunately, though, trying to change this approach by using the two-way binding didn't make a difference.

This is the sequence of events:
1) focus the tag-input, dropdown appears
2) select an item, it gets added, the dropdown disappears
3) the tag-input automatically gets focused again, and the overlay appears again

I've also added a few callbacks to your event hooks, more specifically onBlur, onFocus and onAdd, that simply log a string to the console, and this is the output from the sequence of actions I previously described:
focus (I clicked on the tag-input component)
blur (I moved my mouse on one of the options)
add (I clicked on one of the options)
focus (this happened on its own, and after it the overlay appeared again)
blur (I clicked on the overlay to make it disappear)

**Comment by szanella** [comment]
It's also weird that only the overlay appears, with no dropdown

**Comment by Gbuomprisco** [comment]
And are you also able to reproduce it on the demo page?

**Comment by szanella** [comment]
I did try it on your demo page, but none of that happened. What version of ngx-chips is installed on the demo page?

**Comment by szanella** [comment]
The behaviour there is even different: when I add a new tag, the input gets focused again but neither the dropdown nor the overlay get opened; clicking on the input more doesn't open it either, I need to click out and click in again.
Instead, when I remove a tag, the dropdown opens.

It looks like a whole different version from the one I'm using on my app, which by the way is the 1.4.6

**Comment by Gbuomprisco** [comment]
Yeah it's a different version, just wanted to make sure it's a regression. Would you mind opening a new ticket? The description is very detailed, thanks

**Comment by szanella** [comment]
Sure, I'll paste some of the text I wrote here as a description

Thanks for the assistance!

**Comment by szanella** [comment]
#508
